remote-processor/CMakeLists.txt

```diff
@@ -26,21 +26,20 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-add_library(remote-processor SHARED
+add_library(remote-processor STATIC
         Socket.cpp
         ListeningSocket.cpp
         ConnectionSocket.cpp
         Message.cpp
         RequestMessage.cpp
         AnswerMessage.cpp
-        RemoteProcessorServer.cpp
-        RemoteProcessorServerBuilder.cpp)
+        RemoteProcessorServer.cpp)
 
-set(CMAKE_THREAD_PREFER_PTHREAD 1)
-find_package(Threads REQUIRED)
+if (NOT WIN32)
+    # '-fPIC' needed for linking against shared libraries
```

Comment:
If I'm not mistaken, it's the other way around: it is needed for shared libs to link against this lib.